### PR TITLE
fix: handle edge case for empty disclosures in SdJwtVP parsing

### PR DESCRIPTION
**Pull request by Ogenbertrand**
Here are the necessary changes i made in this file,;

- I added a check to see if disclosureEnd is greater than disclosureStart. If it is, I extract the disclosures string from the original SD-JWT string using substring(disclosureStart + 1, disclosureEnd).

- If disclosureEnd is not greater than disclosureStart, I set disclosuresString to an empty string (""). This ensures that an empty value is assigned when there are no disclosures.

I'll be waiting to for your review @IngridPuppet  and any for any updates on the file.

### CODE REVIEW

File: src/main/java/de/adorsys/sdjwt/vp/SdJwtVP.java [hunk 1]
```diff
@@ -84,7 +84,11 @@ public static SdJwtVP of(String sdJwtString) {
         int disclosureEnd = sdJwtString.lastIndexOf(SdJwt.DELIMITER);
 
         String issuerSignedJWTString = sdJwtString.substring(0, disclosureStart);
-        String disclosuresString = sdJwtString.substring(disclosureStart + 1, disclosureEnd);
+        String disclosuresString = "";
```
Comment (IngridPuppet) [inline src/main/java/de/adorsys/sdjwt/vp/SdJwtVP.java:87]: You'll have a similar problem here if the search on L83 is unsuccessful, returning -1.

**Review by IngridPuppet — CHANGES_REQUESTED** [review]
Can you add tests to check these changes?

File: src/main/java/de/adorsys/sdjwt/vp/SdJwtVP.java (hunk 1 above)
Comment (Ogenbertrand) [reply src/main/java/de/adorsys/sdjwt/vp/SdJwtVP.java:87]: Okay, Let me got through it.

**Comment by Ogenbertrand** [comment]
Hello @IngridPuppet !

i wrote the test to test the modifications made in the SwJwtVP class, and also handle the case when the search on L83 is -1.

how my tests works is;
- Tests the of method with a valid input string, verifying that the resulting SdJwtVP instance has the expected properties.
- And also tests the of method with an input string that lacks a disclosure, verifying that it throws an IllegalArgumentException with the expected message.

[test: add unit tests for SdJwtVP.of(String) method](https://github.com/adorsys/sd-jwt/pull/16/commits/f489e3cd015f320c66270cf0d83089358f3feec8)

File: src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java [hunk 2]
```diff
@@ -176,7 +180,8 @@ public void testS6_2_PresentationPartialDisclosure() throws ParseException, JOSE
         JsonNode keyBindingClaims = TestUtils.readClaimSet(getClass(), "sdjwt/s6.2-key-binding-claims.json");
         // disclose only the given_name
         String presentation = sdJwtVP.present(List.of("jsu9yVulwQQlhFlM_3JlzMaSFzglhQG0DpfayQwLUK4"),
-                keyBindingClaims, testSettings.holderSigContext.signer, testSettings.holderSigContext.keyId, testSettings.jwsAlgorithm, jwsType);
+                keyBindingClaims, testSettings.holderSigContext.signer, testSettings.holderSigContext.keyId,
+                testSettings.jwsAlgorithm, jwsType);
```
Comment (IngridPuppet) [inline src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java:184]: Avoid making noise with formatting.

File: src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java [hunk 3]
```diff
@@ -186,4 +189,21 @@ public void testS6_2_PresentationPartialDisclosure() throws ParseException, JOSE
...
+    assertNotNull(sdJwtVP);
+    assertEquals(4, sdJwtVP.getDisclosures().size());
+    }
+
+    @Test
+    public void testOf_MalformedSdJwt_ThrowsIllegalArgumentException() {
+        // Given
+        String malformedSdJwt = "issuer-signed-jwt"; // missing delimiter at the end
+
+        assertThrows(IllegalArgumentException.class, () -> SdJwtVP.of(malformedSdJwt));
```
Comment (IngridPuppet) [inline src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java:206]: Consider asserting the exception message.

```suggestion

        var exception = assertThrows(IllegalArgumentException.class, () -> SdJwtVP.of(malformedSdJwt));
        
        assertEquals("SD-JWT is malformed, expected to end with ~", exception.getMessage());
```

File: src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java [hunk 4]
```diff
@@ -186,4 +189,21 @@ public void testS6_2_PresentationPartialDisclosure() throws ParseException, JOSE
...
 
-}
+    @Test
+    public void testOf_validInput() {
+    String sdJwtString = TestUtils.readFileAsString(getClass(), "sdjwt/s6.2-presented-sdjwtvp.txt");
+    SdJwtVP sdJwtVP = SdJwtVP.of(sdJwtString);
+
+    assertNotNull(sdJwtVP);
+    assertEquals(4, sdJwtVP.getDisclosures().size());
+    }
```
Comment (IngridPuppet) [inline src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java:199]: Indent the body of the function. When you'll switch to IntelliJ, I'll show you how you can autoformat your changes locally, without affecting the rest of the file.

**Review by IngridPuppet — COMMENTED** [review]
Just a few additional things...

File: src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java (hunk 4 above)
Comment (Ogenbertrand) [reply src/test/java/de/adorsys/sdjwt/sdjwtvp/SdJwtVPTest.java:199]: Okay !
Thank you.

**Comment by Ogenbertrand** [comment]
Okay !
I just updated the test file again to suit the requests. 

[Address reviewer comment: indent method body in testOf_validInput()](https://github.com/adorsys/sd-jwt/pull/16/commits/95911f9bbd382561ad3178670cf290db70b0cb90)

**Review by IngridPuppet — APPROVED** [review]

**Pull request merged**